stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/NonMemberChatEventHandler.kt

```diff
@@ -65,19 +62,11 @@ public class NonMemberChatEventHandler(
     override fun handleMemberRemovedEvent(
         event: MemberRemovedEvent,
         filter: FilterObject,
+        cachedChannel: Channel?,
     ): EventHandlingResult {
-        val channel = runBlocking {
-            val request = channelFilter.filter(client, event.cid, filter)
-            if (request.isSuccess) {
-                request.data().find { channel -> channel.cid == event.cid }
-            } else {
-                null
-            }
-        }
-
         val hasChannel = channels.value.any { it.cid == event.cid }
 
-        return if (!hasChannel && channel != null) EventHandlingResult.Add(channel) else EventHandlingResult.Skip
+        return if (!hasChannel && cachedChannel != null) EventHandlingResult.Add(cachedChannel) else EventHandlingResult.Skip
```

Comment:
If the cachedChannel is null, then it won't add it. If the cache doesn't have that channel, it won't be added. It is necessary to query the channel in the backend to add it when the cache doesn't have the channel